Core/clim-basic/coordinates.lisp

```diff
@@ -68,4 +68,6 @@
 (defun coordinate/= (x y)
   (/= x y))
 
+(deftype rectangle-coordinate-vector () '(simple-array coordinate (4)))
```

Comment:
do you mean a rectangle or a standard-rectangle? the difference is that the rectangle is not required to be aligned with XY (and it is a protocol class) while standard-rectangle is guaranteed to be aligned that way.